app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -174,6 +184,21 @@ public void updateContact(final Contact oldContact, final Contact newContact) th
      * {@inheritDoc}
      */
     @Override
+    public Contact getContactWithName(final String contactName) throws IOException {
+        final List<Contact> contacts = getAllContacts();
+        for (final Contact contact: contacts) {
+            if(contact.getName().equals(contactName)) {
+                return contact;
+            }
+        }
+
+        return null;
```

Comment:
While making it realized that it is not such a nice method to throw an Exception (Because of all the try/catch statements and the other conflicting Exceptions), so I kept it at returning null.